t/08-autotest.t

```diff
@@ -86,88 +84,99 @@ stderr_like { autotest::run_all } qr/finished/, 'run_all outputs status on stder
+        is $reverts_done, 0, 'no snapshots loaded if snapshots are not supported';
+        is $snapshots_made, 0, 'no snapshots made if snapshots are not supported';
+        $reverts_done = $snapshots_made = 0;
+        @sent = ();
+    };
+    subtest 'snapshot loading triggered even when tests successful' => sub {
+        $mock_basetest->redefine(test_flags => {always_rollback => 1});
+        $mock_autotest->redefine(query_isotovideo => 1);
+        stderr_like { autotest::run_all } qr/finished/, 'run_all outputs status on stderr';
+        ($died, $completed) = get_tests_done;
+        is $died, 0, 'start+next+start should not die when always_rollback flag is set';
+        is $completed, 1, 'start+next+start should complete when always_rollback flag is set';
+        is $reverts_done, 0, 'no snapshots loaded if not test with milestone flag';
+        is $snapshots_made, 0, 'no snapshots made if snapshots are not supported';
+        $reverts_done = $snapshots_made = 0;
+        @sent = ();
+    };
+    subtest 'snapshot loading with milestone flag' => sub {
+        $mock_basetest->redefine(test_flags => {always_rollback => 1, milestone => 1});
+        stderr_like { autotest::run_all } qr/finished/, 'run_all outputs status on stderr';
+        ($died, $completed) = get_tests_done;
+        is $died, 0, 'start+next+start should not die when always_rollback flag is set';
+        is $completed, 1, 'start+next+start should complete when always_rollback flag is set';
+        is $reverts_done, 2, 'snapshots are loaded even when tests succeed';
+        is $snapshots_made, 2, 'milestone snapshots are made for all except the last';
+        $reverts_done = $snapshots_made = 0;
+        @sent = ();
+
+        $mock_basetest->redefine(test_flags => {milestone => 1, fatal => 1});
+        stderr_like { autotest::run_all } qr/finished/, 'run_all outputs status on stderr';
+        ($died, $completed) = get_tests_done;
+        is $died, 0, 'start+next+start should not die as fatal milestones';
+        is $completed, 1, 'start+next+start should complete as fatal milestones';
+        is $reverts_done, 0, 'no rollbacks done';
+        is $snapshots_made, 0, 'no snapshots made as no test needed them';
+        $reverts_done = $snapshots_made = 0;
+        @sent = ();
+    };
+    subtest 'stopping overall test execution early due to fatal test failure' => sub {
+        $mock_basetest->redefine(runtest => sub { die "test died\n" });
+        $vm_stopped = 0;
+        stderr_like { autotest::run_all } qr/.*stopping overall test execution after a fatal test failure.*/, 'reason logged';
+        ($died, $completed) = get_tests_done;
+        is $died, 0, 'tests still not considered died if only a test module failed';
+        is $completed, 0, 'tests not considered completed';
+        is $reverts_done, 0, 'no rollbacks done';
+        is $snapshots_made, 0, 'no snapshots made';
+        $reverts_done = $snapshots_made = 0;
+        @sent = ();
```

Comment:
this part is always repeated. Do you think it would work to define something like

```suggestion
sub snapshot_subtest (...) {
        // execute the "sub" reference provided
        // ...
        $reverts_done = $snapshots_made = 0;
        @sent = ();
};

snapshot_subtest 'stopping overall test execution early due to fatal test failure' => sub {
        // subtest instrutions without repeating the same cleanup code repeatedly
};
```

Reply:
awesome!